### PR TITLE
Document `DisableImplicitFrameworkDefines` from the .NET SDK

### LINKED ISSUE #27821: Document the `DisableImplicitFrameworkDefines` SDK Project file property
The SDK defines this property [here](https://github.com/dotnet/sdk/blob/997376c6bc798af22633a8eb36074a0acec8a200/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets#L178-L241) as a marker to determine if it should skip generation of the implicit (generally TFM-and RID-specific) compiler defines (which are gone into in depth [here](https://docs.microsoft.com/en-us/dotnet/standard/frameworks#how-to-specify-a-target-framework).

We should document this property on this page and cross-link to the list of definitions.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cb4428fc-215c-78e1-5ff0-fe73f7a08260
* Version Independent ID: 6741b1de-38a4-fbb1-d79d-acc47d549a2d
* Content: [MSBuild properties for Microsoft.NET.Sdk - .NET](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props)
* Content Source: [docs/core/project-sdk/msbuild-props.md](https://github.com/dotnet/docs/blob/main/docs/core/project-sdk/msbuild-props.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by baronfel**
## Summary

Covers the logic of the property with examples, linking to relevant documentation of the Target Framework concept.

Fixes #27821

### CODE REVIEW

**Comment by baronfel** [comment]
@dsplaisted mind doing a quick once-over for content-correctness?

File: docs/standard/frameworks.md [hunk 1]
```diff
@@ -168,7 +168,7 @@ public class MyClass
 }
 ```
 
-The build system is aware of preprocessor symbols representing the target frameworks shown in the [Supported target framework versions](#supported-target-frameworks) table when you're using SDK-style projects. When using a symbol that represents a .NET Standard, .NET Core, or .NET 5 TFM, replace dots and hyphens with an underscore and change lowercase letters to uppercase (for example, the symbol for `netstandard1.4` is `NETSTANDARD1_4`).
+The build system is aware of preprocessor symbols representing the target frameworks shown in the [Supported target framework versions](#supported-target-frameworks) table when you're using SDK-style projects. When using a symbol that represents a .NET Standard, .NET Core, or .NET 5 TFM, replace dots and hyphens with an underscore and change lowercase letters to uppercase (for example, the symbol for `netstandard1.4` is `NETSTANDARD1_4`). Generation of these symbols can be disabled via the `DisableImplicitFrameworkDefines` property. For more information about this property, see the [documentation](../../docs/core/project-sdk/msbuild-props.md#disableimplicitframeworkdefines).
```
Comment (gewarren) [inline docs/standard/frameworks.md:171]:
```suggestion
The build system is aware of preprocessor symbols representing the target frameworks shown in the [Supported target framework versions](#supported-target-frameworks) table when you're using SDK-style projects. When using a symbol that represents a .NET Standard, .NET Core, or .NET 5+ TFM, replace dots and hyphens with an underscore, and change lowercase letters to uppercase (for example, the symbol for `netstandard1.4` is `NETSTANDARD1_4`). You can disable generation of these symbols via the `DisableImplicitFrameworkDefines` property. For more information about this property, see [DisableImplicitFrameworkDefines](../core/project-sdk/msbuild-props.md#disableimplicitframeworkdefines).
```

File: docs/core/project-sdk/msbuild-props.md [hunk 2]
```diff
@@ -357,6 +358,38 @@ The following MSBuild properties are documented in this section:
 
 C# compiler options, such as `LangVersion` and `Nullable`, can also be specified as MSBuild properties in your project file. For more information, see [C# compiler options](../../csharp/language-reference/compiler-options/index.md).
 
+### DisableImplicitFrameworkDefines
+
+The `DisableImplicitFrameworkDefines` property controls whether or not the SDK generates compiler define symbols for the Target Framework and Platform for the .NET project. When this property is set to `true`, compiler define symbols are generated for
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:363]:
```suggestion
The `DisableImplicitFrameworkDefines` property controls whether or not the SDK generates compiler define symbols for the target framework and platform for the .NET project. When this property is set to `true`, compiler define symbols are generated for:
```

File: docs/core/project-sdk/msbuild-props.md [hunk 3]
```diff
@@ -357,6 +358,38 @@ The following MSBuild properties are documented in this section:
...
+
+* Framework without version (`NETFRAMEWORK`, `NETSTANDARD`, `NET`)
+* Framework with version (`NET48`, `NETSTANDARD2_0`, `NET6_0`)
+* Framework with version minimum bound (`NET48_OR_GREATER`, `NETSTANDARD2_0_OR_GREATER`, `NET6_0_OR_GREATER`)
+
+For more information on Target Framework Monikers and these implicit compiler define symbols, see the [Target frameworks](../../standard/frameworks.md) documentation.
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:369]:
```suggestion
For more information on target framework monikers and these implicit compiler define symbols, see [Target frameworks](../../standard/frameworks.md).
```

File: docs/core/project-sdk/msbuild-props.md [hunk 4]
```diff
@@ -357,6 +358,38 @@ The following MSBuild properties are documented in this section:
...
+
+Additionally, if an operating system-specific Target Framework is specified in the project (for example `net6.0-android`), the following compiler define symbols are generated:
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:371]:
```suggestion
Additionally, if you specify an operating system-specific target framework in the project (for example `net6.0-android`), the following compiler define symbols are generated:
```

File: docs/core/project-sdk/msbuild-props.md [hunk 5]
```diff
@@ -357,6 +358,38 @@ The following MSBuild properties are documented in this section:
...
+
+* Platform without version (`ANDROID`, `IOS`, `WINDOWS`)
+* Platform with version (`IOS15_1`)
+* Platform with version minimum bound (`IOS15_1_OR_GREATER`)
+
+For more information on operating system-specific Target Framework Monikers, see the documentation on [OS-specific TFMs](../../standard/frameworks.md#net-5-os-specific-tfms).
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:377]:
```suggestion
For more information on operating system-specific target framework monikers, see [OS-specific TFMs](../../standard/frameworks.md#net-5-os-specific-tfms).
```

File: docs/core/project-sdk/msbuild-props.md [hunk 6]
```diff
@@ -357,6 +358,38 @@ The following MSBuild properties are documented in this section:
...
+
+Finally, if your Target Framework implies support for older Target Frameworks, compiler define symbols for those older frameworks will be emitted. For example, `net6.0` implies support for `net5.0` and so on all the way back to `.netcoreapp1.0`, and so for each of these Target Frameworks, the _Framework with version_ and _Framework with version minimum bound_ symbols will be defined.
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:379]:
```suggestion
Finally, if your target framework implies support for older target frameworks, compiler define symbols for those older frameworks are emitted. For example, `net6.0` implies support for `net5.0` and so on all the way back to `.netcoreapp1.0`. So for each of these target frameworks, the _Framework with version_ and _Framework with version minimum bound_ symbols will be defined.
```

**Review by gewarren — COMMENTED** [review]
Thanks for adding this. One overall comment: should we refer to them as "compiler define symbols" or "preprocessor symbols"? I think that "compiler define symbols" might be hard to translate.

**Comment by baronfel** [comment]
@gewarren good idea, I've swapped that terminology in the new content.

**Review by gewarren — APPROVED** [review]
LGTM!

File: docs/core/project-sdk/msbuild-props.md [hunk 7]
```diff
@@ -357,6 +358,38 @@ The following MSBuild properties are documented in this section:
 
 C# compiler options, such as `LangVersion` and `Nullable`, can also be specified as MSBuild properties in your project file. For more information, see [C# compiler options](../../csharp/language-reference/compiler-options/index.md).
 
+### DisableImplicitFrameworkDefines
```
[the remark this answers is not in the document]
Comment (baronfel) [reply docs/core/project-sdk/msbuild-props.md:361]: good spot, added in the latest version.

File: docs/core/project-sdk/msbuild-props.md [hunk 8]
```diff
@@ -357,6 +358,38 @@ The following MSBuild properties are documented in this section:
...
+
+Additionally, if you specify an operating system-specific target framework in the project (for example `net6.0-android`), the following preprocessor symbols are generated:
+
+* Platform without version (`ANDROID`, `IOS`, `WINDOWS`)
+* Platform with version (`IOS15_1`)
+* Platform with version minimum bound (`IOS15_1_OR_GREATER`)
+
+For more information on operating system-specific target framework monikers, see [OS-specific TFMs](../../standard/frameworks.md#net-5-os-specific-tfms).
+
+Finally, if your target framework implies support for older target frameworks, preprocessor symbols for those older frameworks are emitted. For example, `net6.0` implies support for `net5.0` and so on all the way back to `.netcoreapp1.0`. So for each of these target frameworks, the _Framework with version_ and _Framework with version minimum bound_ symbols will be defined.
```
Comment (Youssef1313) [inline docs/core/project-sdk/msbuild-props.md:379]: Is the "dot" extra?

```suggestion
Finally, if your target framework implies support for older target frameworks, preprocessor symbols for those older frameworks are emitted. For example, `net6.0` implies support for `net5.0` and so on all the way back to `netcoreapp1.0`. So for each of these target frameworks, the _Framework with version_ and _Framework with version minimum bound_ symbols will be defined.
```

**Review by Youssef1313 — APPROVED** [review]

**Review by dsplaisted — APPROVED** [review]

File: docs/core/project-sdk/msbuild-props.md [hunk 9]
```diff
@@ -357,6 +358,38 @@ The following MSBuild properties are documented in this section:
...
+
+Additionally, if you specify an operating system-specific target framework in the project (for example `net6.0-android`), the following preprocessor symbols are generated:
+
+* Platform without version (`ANDROID`, `IOS`, `WINDOWS`)
+* Platform with version (`IOS15_1`)
+* Platform with version minimum bound (`IOS15_1_OR_GREATER`)
+
+For more information on operating system-specific target framework monikers, see [OS-specific TFMs](../../standard/frameworks.md#net-5-os-specific-tfms).
+
+Finally, if your target framework implies support for older target frameworks, preprocessor symbols for those older frameworks are emitted. For example, `net6.0` implies support for `net5.0` and so on all the way back to `netcoreapp1.0`. So for each of these target frameworks, the _Framework with version_ and _Framework with version minimum bound_ symbols will be defined.
```
[the remark this answers is not in the document]
Comment (baronfel) [reply docs/core/project-sdk/msbuild-props.md]: thanks, removed from that paragraph.

**Comment by baronfel** [comment]
Merging because of approvals. Thanks all!

**Pull request merged**